lib/nostrum/api.ex

```diff
@@ -1355,54 +1360,82 @@ defmodule Nostrum.Api do
   end
 
   @doc """
-  Gets a user.
+  Gets a user by its `user_id`.
 
-  User to get is specified by `user_id`.
+  If the request is successful, this function returns `{:ok, user}`, where 
+  `user` is a `Nostrum.Struct.User`. Otherwise, returns `{:error, reason}`.
   """
-  @spec get_user(integer) :: error | {:ok, Nostrum.Sturct.User.t}
+  @spec get_user(User.id) :: error | {:ok, User.t}
   def get_user(user_id) do
     request(:get, Constants.user(user_id))
+    |> handle_request_with_decode({:struct, User})
   end
 
-  @spec get_user!(integer) :: no_return | Nostrum.Struct.User.t
+  @doc """
+  Same as `get_user/1`, but raises `Nostrum.Error.ApiError` in case of failure.
+  """
+  @spec get_user!(User.id) :: no_return | User.t
   def get_user!(user_id) do
     get_user(user_id)
     |> bangify
   end
 
   @doc """
   Gets info on the current user.
+
+  If nostrum's caching is enabled, it is recommended to use `Nostrum.Cache.Me.get/0` 
+  instead of this function. This is because sending out an API request is much slower 
+  than pulling from our cache.
+
+  If the request is successful, this function returns `{:ok, user}`, where 
+  `user` is nostrum's `Nostrum.Struct.User`. Otherwise, returns `{:error, reason}`.
   """
-  @spec get_current_user() :: error | {:ok, Nostrum.Struct.User.t}
+  @spec get_current_user() :: error | {:ok, User.t}
   def get_current_user do
-    case request(:get, Constants.me) do
-      {:ok, body} ->
-        {:ok, Poison.decode!(body, as: %User{})}
-      other ->
-        other
-    end
+    request(:get, Constants.me)
+    |> handle_request_with_decode({:struct, User})
+  end
+
+  @doc """
+  Same as `get_current_user/0`, but raises `Nostrum.Error.ApiError` in case of failure.
+  """
+  @spec get_current_user!() :: no_return | User.t
+  def get_current_user! do
+    get_current_user()
+    |> bangify
   end
 
   @doc ~S"""
   Changes the username or avatar of the current user.
 
-  **Example**
-  ```Elixir
-  avatar = %{avatar: "data:image/jpeg;base64," <> "YXl5IGJieSB1IGx1a2luIDQgc3VtIGZ1az8="}
-  {:ok, user} = Nostrum.Api.modify_current_user(avatar)
-  ```
+  ## Params 
+ 
+  The following params are optional: 
+ 
+    * `:username` (string) - new username
```

Comment:
This kind of `param` documentation is missing on the other functions. We should discuss if we should include it everywhere, or only when accepting lists/maps and optional parameters. 

I'm leaning towards the latter, but the former could be helpful if the user isn't familiar with the API overall. But this could probably be handled by with types and such. Let me know what you think.

Reply:
Agreed with the latter. Was planning to change the rest of the API functions to only include a "Request Params" section whenever lists/maps were needed.